packages/editor/src/pages/External/index.tsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
 
-import { parse } from 'query-string';
 import safeExternalUrls from 'common/lib/safe.external.urls';
 
 import { RunOnLoad } from 'common/lib/components/PageSwitcher/utilities/RunOnLoad';
 
 function setup() {
-  const { destination } = parse(window.location.search) as {
-    destination: string;
-  };
+  // Note: using just an indexOf of whatever follows "#/external-page?destination="
+  //    rather than doping an actual search string query.
+  //    This is because, as part of launching the dialog, Office.js prepends a bunch
+  //    of stuff on the URL as a query string (?_host_Info=) BEFORE the hash,
+  //    and doesn't seem to want a URL with an existing query string in there.
+  //    On the routing side, in turn, it appears that having the destination
+  //    embedded directly after the "#/external-page/<encoded url>"
+  //    isn't working either.  So just end up with two "?" on the URL,
+  //    which -- though weird -- seems to work just fine.
```

Comment:
NIT: this comment is a little confusing. I read it a few times before I really got what you were saying. Maybe try summarizing a bit?

Reply:
Ok, shortened it.  It's a really strange issue, but I think in practice as long as it works, no one should need to touch it again...